### PR TITLE
Flagger addon

### PR DESCRIPTION
**Pull request by Eli1123**
*Issue #, if available:*

*Description of changes:*
working on flagger now and starting new pull request

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/flagger/my_new_addon.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// filename=lib/kubevious_addon.ts
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:1]: remove this line

File: lib/addons/flagger/my_new_addon.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import 'source-map-support/register';
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:2]: please use consistent naming for files. this is not a hello world project, it is real production level code. should be `index.ts`.

File: lib/addons/flagger/my_new_addon.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * populateValues populates the appropriate values used to customize the Helm chart
+ * @param helmOptions User provided values to customize the chart
+ */
+function populateValues(helmOptions: FlaggerAddOnProps): blueprints.Values {
+  const values = helmOptions.values ?? {};
+
+  setPath(values, "ingress.enabled", helmOptions.ingressEnabled);
+  setPath(values, "flagger.service.type", helmOptions.flaggerServiceType);
+  setPath(values, "mysql.generate_passwords", true);
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:62]: what is this for?

File: lib/addons/flagger/my_new_addon.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { Construct } from 'constructs';
+import { setPath } from '../../utils';
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface FlaggerAddOnProps extends blueprints.HelmAddOnUserProps {
+  version?: string,
+  ingressEnabled?: boolean,
+  flaggerServiceType?: string,
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:13]: if it allows only specific values, it should be an enum

File: lib/addons/flagger/my_new_addon.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import * as blueprints from '../../../lib';
+import { Construct } from 'constructs';
+import { setPath } from '../../utils';
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface FlaggerAddOnProps extends blueprints.HelmAddOnUserProps {
+  version?: string,
+  ingressEnabled?: boolean,
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:12]: what does it control? need docs

File: lib/addons/flagger/my_new_addon.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import 'source-map-support/register';
+import * as blueprints from '../../../lib';
+import { Construct } from 'constructs';
+import { setPath } from '../../utils';
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface FlaggerAddOnProps extends blueprints.HelmAddOnUserProps {
+  version?: string,
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:11]: don't use comma to delineate interface members, it is a bad practice. You must use `;` and clear documentation on each member

Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:11]: why version is here? is it not in the base interface already?

File: lib/addons/flagger/my_new_addon.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  ingressEnabled?: boolean,
+  flaggerServiceType?: string,
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+export const defaultProps: blueprints.HelmAddOnProps & FlaggerAddOnProps = {
+  name: "flagger-progressive-delivery", // Internal identifyer for our add-on
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:20]: this addon is `flagger`, let's call it that.

File: lib/addons/flagger/my_new_addon.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * populateValues populates the appropriate values used to customize the Helm chart
+ * @param helmOptions User provided values to customize the chart
+ */
+function populateValues(helmOptions: FlaggerAddOnProps): blueprints.Values {
+  const values = helmOptions.values ?? {};
+
+  setPath(values, "ingress.enabled", helmOptions.ingressEnabled);
+  setPath(values, "flagger.service.type", helmOptions.flaggerServiceType);
```
Comment (shapirov103) [inline lib/addons/flagger/my_new_addon.ts:61]: I don't see anything related to prometheus. Why is it not an member of the props object?

**Review by shapirov103 — CHANGES_REQUESTED** [review]
Please remove the commits from the other PR, it creates a lot of noise and makes it hard to review. You can close the PR if needed and drop the branch, recreate and move the one file back.

File: lib/addons/flagger/my_new_addon.ts (hunk 2 above)
Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:2]: I assume this refers to the name my_new_addon.ts that should be index.ts, my bad.

File: lib/addons/flagger/my_new_addon.ts (hunk 3 above)
Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:62]: Example from outline I did not remove yet

File: lib/addons/flagger/my_new_addon.ts (hunk 4 above)
Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:13]: Example from outline I did not remove yet. If I need these later I'll keep that in mind for the design.

File: lib/addons/flagger/my_new_addon.ts (hunk 5 above)
Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:12]: Example from outline I did not remove yet

File: lib/addons/flagger/my_new_addon.ts (hunk 6 above)
Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:11]: Example from outline I did not remove yet, but that is good to keep in mind.

Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:11]: Example from outline I did not remove yet.

File: lib/addons/flagger/my_new_addon.ts (hunk 7 above)
Comment (Eli1123) [reply lib/addons/flagger/my_new_addon.ts:20]: Felt weird keeping name, namespace, and chart all "flagger" but I guess that is the proper naming scheme.

Comment (shapirov103) [reply lib/addons/flagger/my_new_addon.ts]: There are cases when these can be different. For example if the same addon is added more than once to the blueprint. another case: containerinsights installes adot collector helm chart. However, in most cases they are identical.

**Comment by shapirov103** [comment]
> Please remove the commits from the other PR, it creates a lot of noise and makes it hard to review. You can close the PR if needed and drop the branch, recreate and move the one file that you created back.

@Eli1123 do you mind addressing this ^^ one first? also merge from main.

**Pull request closed without merging**